lib/Backend/PreLowerPeeps.cpp

```diff
@@ -98,6 +98,14 @@ IR::Instr *Lowerer::PeepShl(IR::Instr *instrShl)
     instrDef->Remove();
 
     IntConstType oldValue = src2->AsIntConstOpnd()->GetValue();
+
+    // Left shift operator (<<) on arm32 is implemented by LSL which doesn't discard bits beyond lowerest 5-bit.
```

Comment:
Tom's point is that the (1 << oldValue) behaves differently on x86/x64 vs ARM.  x86/x64 only use the 5 bits of the shift count as ARM effectively uses all the bits.  JavaScript says to only use the bottom 5 bits, so the x86/x64 behavior is correct.  

For code clarity however, I would remove the #ifdef.

---
In reply to: [100117486](https://github.com/Microsoft/ChakraCore/pull/2486#discussion_r100117486) [](ancestors = 100117486)